esx_service/utils/vmdk_utils.py

```diff
@@ -331,9 +336,27 @@ def get_datastore_name(datastore_url):
         return auth.DEFAULT_DS
 
     # Query datastore name from VIM API
+    logging.debug("get_datastore_name: ds_objects=%s", get_datastore_objects())
     res = [d.info.name for d in get_datastore_objects() if d.info.url == datastore_url]
     return res[0]
 
+def get_datastore_name_from_config_path(config_path):
+    """Returns datastore NAME in config_path (not url-name which may be used in path)
+    """
+    # path is always /vmfs/volumes/<datastore>/... , so extract datastore:
+    config_ds_name = config_path.split("/")[3]
+    # get_datastore() return the cached value
+    # if datastore is changed outside
+    # ds_name may have a stale value
+    # therefore, force to refresh datastore cache by calling get_datastores()
+    # with need_refresh=True
+    ds_name = [x[0] for x in get_datastores(need_refresh=True) \
+                    if x[0] == config_ds_name or x[1] == config_ds_name ]
+    if len(ds_name) != 1:
+        logging.error("get_datastore_name_from_config_path: found more than one match: %s" % ds_name)
```

Comment:
I see the code block is just a move from vmdk_ops .. Is there any plan to get rid off duplicates from the cache value?